app-authz-spring-security/README.md

```diff
@@ -0,0 +1,97 @@
+app-authz-spring-security: Spring Security Application using Authorization Services
+===================================================
+
+Level: Beginner
+Technologies: SpringBoot, Spring MVC, Spring Security
+Summary: SpringBoot, Spring MVC, Spring Security
+Target Product: <span>Keycloak</span>
+Source: <https://github.com/keycloak/keycloak-quickstarts>
+
+
+What is it?
+-----------
+
+The `app-authz-spring-security` quickstart demonstrates how to write a Spring Security application where both authentication and
+authorization aspects are managed by <span>Keycloak</span>.
+
+This application tries to focus on the authorization features provided by <span>Keycloak</span> Authorization Services, where resources are
+protected by a set of permissions and policies defined in Keycloak itself and access to these resources are enforced by a policy enforcer
+that intercepts every single request to the application.
+
+In this application, there are three paths protected by specific permissions in <span>Keycloak</span>:
+
+* **/protected**, where access to this page is based on the evaluation of permissions associated with a resource **Protected Resource** in <span>Keycloak</span>. Basically,
+any user with a role *user* is allowed to access this page.
+
+* **/protected/premium**, where access to this page is based on the evaluation of permissions associated with a resource **Premium Resource** in <span>Keycloak</span>. Basically,
+only users with a role *user-premium* is allowed to access this page.
+
+* **/protected/alice**, where access to this page is based on the evaluation of permissions associated with a resource **Alice Resource** in <span>Keycloak</span>. Basically,
+only user *alice* is allowed to access this page.
+
+The home page (home.ftl) also demonstrates how to use a ``AuthorizationContext`` instance to check for user`s permissions and hide/show
+things in a page. Where the ``AuthorizationContext`` encapsulates all permissions granted by a <span>Keycloak</span> server and provides methods
+to check these permissions.
+
+You can use two distinct users to access this application:
+
+|Username|Password|Roles|
+|---|---|---|
+|alice|alice|user|
+|jdoe|jdoe|user, user-premium|
+
+
+System Requirements
+-------------------
+
+All you need to build this project is Java 8.0 (Java SDK 1.8) or later and Maven 3.1.1 or later.
+
+
+Configuration in <span>Keycloak</span>
+-----------------------
+
+Prior to running the quickstart you need to create a `realm` in <span>Keycloak</span> with all the necessary configuration to deploy and run the quickstart.
+
+The following steps show how to create the realm required for this quickstart:
```

Comment:
I suggest this as a standard introduction to a procedure:
To create the realm required for this quickstart, follow these steps:

Reply:
Oh, also, use an ordered list, with numbers, instead of the bullets.